src/components/FeatureViews/Notes/Notes.js

```diff
@@ -15,503 +17,462 @@ import EditIcon from "@material-ui/icons/EditOutlined";
+                        <NotificationIcon className="notification"
+                            onClick={toggleNotification}
+                            style={notificationColor} />
+                    </Tooltip>
+                </DialogTitle>
+                <DialogContent>
+                    <InputBase className="note-body"
+                        inputProps={{"aria-label": "naked"}} multiline
+                        onChange={handleBodyUpdate}
+                        placeholder="Body (required)" required rows={15}
+                        value={noteBody} variant="filled" />
+                </DialogContent>
+                <DialogActions>
+                    <Button onClick={hideWarning} variant="outlined">
+                        Cancel
+                    </Button>
+                    <Button color="primary" disabled={!noteBody}
+                        onClick={saveNote} variant="outlined">
+                        {submitting ? "Saving..." : "Save"}
+                    </Button>
+                    {!submitting && submitError &&
+                        <span style={{"float": "right"}}>
+                            Error while saving!
+                        </span>}
+                </DialogActions>
+            </Dialog>
+            <Dialog aria-describedby="simple-modal-description"
+                aria-labelledby="simple-modal-title" className="delete-popup"
+                fullWidth maxWidth="xs" onClose={hideWarning}
+                open={deleteID !== null}>
+                <DialogTitle>Confirm Delete</DialogTitle>
+                <DialogContent>
+                    Are you sure you want to delete {getNoteByID(deleteID) &&
+                        getNoteByID(deleteID).title ?
+                        `"${getNoteByID(deleteID).title}"` :
+                        "this note"}?
+                </DialogContent>
+                <DialogActions className="delete-actions">
+                    <Button className="cancel-button" onClick={hideWarning}
+                        variant="contained">
+                        Cancel
+                    </Button>
+                    <Button className="delete-button" onClick={handleDelete}
+                        variant="contained">
+                        Delete
+                    </Button>
+                    {deleteError &&
+                        <span style={{"float": "right"}}>
+                            Error while deleting!
+                        </span>}
+                </DialogActions>
+            </Dialog>
+            <Grid item md={3}>
+                <div className="addNote" onClick={openNewNote}
+                    style={{"cursor": "pointer"}}>
+                    <Typography className="center">
+                        <AddIcon /><br />Add Note
+                    </Typography>
+                </div>
+            </Grid>
+            {notes && Object.values(notes).map((note) => (
```

Comment:
is the `notes &&` necessary now that we're waiting for the `loading` var to be true?